sdk/core-rest/core-client/src/urlHelpers.ts

```diff
@@ -0,0 +1,42 @@
+import { RequestParameters } from "./pathClientTypes";
+
+/**
+ * Builds the request url, filling in query and path parameters
+ * @param baseUrl base url which can be a template url
+ * @param routePath path to append to the baseUrl
+ * @param pathParameters values of the path parameters
+ * @param options request parameters including query parameters
+ * @returns a full url with path and query parameters
+ */
+export function buildRequestUrl(
+  baseUrl: string,
+  routePath: string,
+  pathParameters: string[],
+  options: RequestParameters = {}
+): string {
+  let path = routePath;
+
+  if (path.startsWith("https://") || path.startsWith("http://")) {
+    return path;
+  }
+
+  for (const pathParam of pathParameters) {
+    path = path.replace(/{([^\/]+)}/, pathParam);
+  }
+
+  const url = new URL(`${baseUrl}/${path}`);
+
+  if (options.queryParameters) {
+    const queryParams = options.queryParameters;
+    for (const key of Object.keys(queryParams)) {
+      url.searchParams.append(key, (queryParams as any)[key]);
+    }
+  }
+
+  return (
+    url
+      .toString()
+      // Remove double forward slashes
```

Comment:
when does this happen?

Reply:
This happens when the endpoint has an '/' at the end and we append the path which always starts with / eg. (/foo/bar)

Reply:
Ah. I'm not opposed to this approach, but since regex is tricky to reason about, I would consider simply doing endpoint.endsWith("/") and slicing it off. As it stands I had to squint hard at the regex to verify that `https://` didn't become `http:/`

Reply:
could this function be used by `core-client`'s `getRequestUrl`? I guess we will have to downlevel from `OperationSpec` to the inputs this function expects.